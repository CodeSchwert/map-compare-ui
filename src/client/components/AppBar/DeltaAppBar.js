import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import { Typography } from "@material-ui/core";
import { AppBar, Toolbar } from "@material-ui/core";
import Logo from "./../../../static/Logo.png";
import DateSelect from "../DateSelect/DateSelect";
import ProjectSelect from "../ProjectSelect/ProjectSelect";

import { AppContext } from "../../contexts/AppContext";

const styles = theme => ({
  root: {
    flexGrow: 1,
  },
  flex: {
    flex: 1,
  },
  appBar: {
    backgroundColor: theme.palette.primary.dark,
    padding: "10px 0px"
  },
  toolbar: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  img: {
    alignSelf: "center",
    justifySelf: "right",
    height: "70px"
  },
  title: {
    alignSelf: "center"
  },
  buttonGroup: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-around"
  },
  appBarButton: {
    padding: "5px"
  }
});

class DeltaAppBar extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  static propTypes = {
    classes: PropTypes.object.isRequired,
    handleDateUpdate: PropTypes.func.isRequired,
    handleProjectUpdate: PropTypes.func.isRequired
  };

  render() {
    const { classes, handleDateUpdate, handleProjectUpdate } = this.props;

    if (handleDateUpdate && handleProjectUpdate) {
      return (
        <div className={classes.root}>
          <AppBar className={classes.appBar}>
            <Toolbar className={classes.toolbar}>
              <img
                className={classes.img}
                src={Logo}
              />
              <AppContext.Consumer>
                {appState => {
                  const { project } = appState;
  
                  return (
                    <Typography
                      className={classes.title}
                      variant="h4"
                      color="inherit"
                    >
                      Map Compare {project.name ? `- ${project.name}` : ''}
                    </Typography>
                  );
                }}
              </AppContext.Consumer>
              <div className={classes.buttonGroup}>
                <DateSelect 
                  className={classes.appBarButton}
                  handleDateUpdate={handleDateUpdate}  
                />
                <div className={classes.appBarButton}>{/* spacer */}</div>
                <ProjectSelect 
                  handleProjectUpdate={handleProjectUpdate}
                  className={classes.appBarButton}
                />
              </div>
            </Toolbar>
          </AppBar>
        </div>
      );
    } else {
      return <div></div>
    }
  }
}

export default withStyles(styles)(DeltaAppBar);
