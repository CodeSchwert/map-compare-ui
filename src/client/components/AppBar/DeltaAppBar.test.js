import React from "react";
import { shallow, render } from "enzyme";
import DeltaAppBar from "./DeltaAppBar";

describe("DeltaAppBar", () => {
  it("should render correctly", () => {
    expect.assertions(1);
    const container = shallow(<DeltaAppBar />);
    expect(container.length).toEqual(1);
  });

  it("should match the snapshot.", () => {
    expect.assertions(1);
    const renderedValue = render(<DeltaAppBar />);
    expect(renderedValue).toMatchSnapshot();
  });
});
