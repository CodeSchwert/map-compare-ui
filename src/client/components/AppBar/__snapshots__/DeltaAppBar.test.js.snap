// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeltaAppBar should match the snapshot. 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-2 MuiAppBar-positionFixed-3 MuiAppBar-colorPrimary-9 mui-fixed VectorList-appBar-1",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "src": "[object Object]",
          "style": "align-self:center;justify-self:right;height:70%",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "MuiTypography-root-38 MuiTypography-h4-53 MuiTypography-colorInherit-67",
            "style": "align-self:center",
          },
          "children": Array [
            Object {
              "data": "Delta",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h4",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "src": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "src": undefined,
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "MuiTypography-root-38 MuiTypography-h4-53 MuiTypography-colorInherit-67",
          "style": "align-self:center",
        },
        "children": Array [
          Object {
            "data": "Delta",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h4",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "src": "[object Object]",
            "style": "align-self:center;justify-self:right;height:70%",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "src": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "src": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "header",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
